starboard/android/shared/android_main.cc

```diff
@@ -249,103 +249,104 @@ void* ThreadEntryPoint(void* context) {
-}
-
-void OnResume(GameActivity* activity) {
-  if (g_app_running.load()) {
-    ApplicationAndroid::Get()->SendAndroidCommand(AndroidCommand::kResume);
-  }
-}
-
-void OnPause(GameActivity* activity) {
-  if (g_app_running.load()) {
-    ApplicationAndroid::Get()->SendAndroidCommand(AndroidCommand::kPause);
-  }
-}
-
-void OnStop(GameActivity* activity) {
-  if (g_app_running.load()) {
-    ApplicationAndroid::Get()->SendAndroidCommand(AndroidCommand::kStop);
-  }
-}
-
-bool OnTouchEvent(GameActivity* activity,
-                  const GameActivityMotionEvent* event) {
-  if (g_app_running.load()) {
-    return ApplicationAndroid::Get()->SendAndroidMotionEvent(event);
-  }
-  return false;
-}
-
-bool OnKey(GameActivity* activity, const GameActivityKeyEvent* event) {
-  if (g_app_running.load()) {
-    return ApplicationAndroid::Get()->SendAndroidKeyEvent(event);
-  }
-  return false;
-}
-
-void OnWindowFocusChanged(GameActivity* activity, bool focused) {
-  if (g_app_running.load()) {
-    ApplicationAndroid::Get()->SendAndroidCommand(
-        focused ? AndroidCommand::kWindowFocusGained
-                : AndroidCommand::kWindowFocusLost);
-  }
-}
-
-void OnNativeWindowCreated(GameActivity* activity, ANativeWindow* window) {
-  g_block_swapbuffers.store(false);
-  if (g_app_running.load()) {
-    ApplicationAndroid::Get()->SendAndroidCommand(
-        AndroidCommand::kNativeWindowCreated, window);
-  }
-}
-
-void OnNativeWindowDestroyed(GameActivity* activity, ANativeWindow* window) {
-  g_block_swapbuffers.store(true);
-  if (g_app_running.load()) {
-    ApplicationAndroid::Get()->SendAndroidCommand(
-        AndroidCommand::kNativeWindowDestroyed);
-  }
-}
-
-extern "C" SB_EXPORT_PLATFORM void GameActivity_onCreate(
-    GameActivity* activity,
-    void* savedState,
-    size_t savedStateSize) {
-  // Start the Starboard thread the first time an Activity is created.
-  if (g_starboard_thread == 0) {
-    Semaphore semaphore;
-
-    pthread_attr_t attributes;
-    pthread_attr_init(&attributes);
-    pthread_attr_setdetachstate(&attributes, PTHREAD_CREATE_DETACHED);
-
-    pthread_create(&g_starboard_thread, &attributes, &ThreadEntryPoint,
-                   &semaphore);
-
-    pthread_attr_destroy(&attributes);
-
-    // Wait for the ApplicationAndroid to be created.
-    semaphore.Take();
-  }
-
-  activity->callbacks->onStart = OnStart;
-  activity->callbacks->onResume = OnResume;
-  activity->callbacks->onPause = OnPause;
-  activity->callbacks->onStop = OnStop;
-  activity->callbacks->onTouchEvent = OnTouchEvent;
-  activity->callbacks->onKeyDown = OnKey;
-  activity->callbacks->onKeyUp = OnKey;
-  activity->callbacks->onWindowFocusChanged = OnWindowFocusChanged;
-  activity->callbacks->onNativeWindowCreated = OnNativeWindowCreated;
-  activity->callbacks->onNativeWindowDestroyed = OnNativeWindowDestroyed;
-
-  activity->instance = ApplicationAndroid::Get();
-}
+// TODO(cobalt): determine if we still need native app lifecycle events.
```

Comment:
I would generally bias on deleting the code that we don't know if we need. It's easy enough to find it in old branches